tests/unit/test_scaffold.py

```diff
@@ -1,88 +1,261 @@
+    assert scaffold.app is not None
+    assert scaffold.app.config.get('GIT_BASE_URL') == url
+    assert (scaffold.app.config.get('CUSTOM_SETTINGS') ==
+            'instance/config.yml')
+    assert (scaffold.app.config.get('default_params').get('a_key') ==
+            'some value')
+
+
+def test_valid_instance_configs(mock_instance_folder):
+    """
+    GIVEN an existing Flask app
+    WHEN it is passed into the Scaffold
+    THEN the default settings in the specified instance folder are loaded
+    """
+    url = 'http://some.site/cgit/plain/additional-tags?h='
+    base_app = Flask('TestApp', instance_path=mock_instance_folder,
+                     instance_relative_config=True)
+    scaffold = AppScaffold(app=base_app)
+    assert scaffold.app is not None
+    assert scaffold.app.config.get('GIT_BASE_URL') == url
+    assert (scaffold.app.config.get('CUSTOM_SETTINGS') ==
+            'instance/config.yml')
+    assert (scaffold.app.config.get('default_params').get('a_key') ==
+            'some value')
 
 
-def test_setting_incorrect_env_var(monkeypatch):
+def test_loads_all_values_in_extra_cfg_file(mock_custom_only_extra_cfg):
     """
-    GIVEN a test mapping is passed to create_app
-    AND the wrong config values are set as env vars
+    GIVEN a config mapping is passed to AppScaffold
+    AND that mapping contains a custom setting pointing to a cfg file
     WHEN we try to create the app
-    THEN it fails to start the app
-    AND throws an error
+    THEN it creates an instance of the app with the expected setting
     """
-    monkeypatch.setenv('WRONG_KEY', 'http://foo.com/cgit')
-    with pytest.raises(ValueError):
-        AppScaffold(config={'TESTING': True})
+    scaffold = AppScaffold(config={
+        'TESTING': True,
+        'CUSTOM_SETTINGS': mock_custom_only_extra_cfg})
+    assert scaffold.app is not None
+    base_custom = scaffold.app.config.get('custom_section')
+    assert base_custom.get('Different_case') == 'val'
+    assert base_custom.get('all_lower') == 'something'
+
+
+def test_fails_on_bad_custom_cfg_format(mock_extra_settings_file):
+    """
+    GIVEN a config mapping is passed to AppScaffold
+    AND that mapping contains a custom setting pointing to an invalid cfg file
+    WHEN we try to create the app
+    THEN it raises a MissingSectionHeaderError
+    """
+    with pytest.raises(configparser.MissingSectionHeaderError):
+        AppScaffold(config={
+            'TESTING': True,
+            'CUSTOM_SETTINGS': mock_extra_settings_file})
+
+
+def test_referenced_config_file_not_found(mock_custom_only_bad_file):
+    """
+    GIVEN a config mapping is passed to AppScaffold
+    AND an invalid file is referenced by the FLASK_SETTINGS env var
+    WHEN we try to create the app
+    THEN it raises a FileNotFoundError
+    """
+    with pytest.raises(FileNotFoundError):
+        AppScaffold(config={'TESTING': True,
+                    'CUSTOM_SETTINGS': mock_custom_only_bad_file})
```

Comment:
The dict indentation looks unusual to me, but if it passes flake8 I suppose it is fine. The one used above looks more natural to me.

Reply:
Same, I just figured out how to get it to pass flake8, uploading in a sec

Reply:
Oh, and I have a task this sprint to fix the mergify issue.